apiserver/facades/client/action/operation.go

```diff
@@ -5,94 +5,170 @@ package action
 
 import (
 	"context"
+	"strings"
 
-	"github.com/juju/errors"
+	jujuerrors "github.com/juju/errors"
 	"github.com/juju/names/v6"
 
 	apiservererrors "github.com/juju/juju/apiserver/errors"
-	coreunit "github.com/juju/juju/core/unit"
-	applicationerrors "github.com/juju/juju/domain/application/errors"
+	"github.com/juju/juju/core/unit"
+	"github.com/juju/juju/domain/operation"
 	"github.com/juju/juju/rpc/params"
+
+	"github.com/juju/juju/internal/errors"
 )
 
 // EnqueueOperation takes a list of Actions and queues them up to be executed as
 // an operation, each action running as a task on the designated ActionReceiver.
 // We return the ID of the overall operation and each individual task.
 func (a *ActionAPI) EnqueueOperation(ctx context.Context, arg params.Actions) (params.EnqueuedActions, error) {
-	operationId, actionResults, err := a.enqueue(ctx, arg)
-	if err != nil {
-		return params.EnqueuedActions{}, err
-	}
-	results := params.EnqueuedActions{
-		OperationTag: names.NewOperationTag(operationId).String(),
-		Actions:      actionResults.Results,
+	if err := a.checkCanWrite(ctx); err != nil {
+		return params.EnqueuedActions{}, errors.Capture(err)
 	}
-	return results, nil
-}
 
-func (a *ActionAPI) enqueue(ctx context.Context, arg params.Actions) (string, params.ActionResults, error) {
-	if err := a.checkCanWrite(ctx); err != nil {
-		return "", params.ActionResults{}, errors.Trace(err)
+	if len(arg.Actions) == 0 {
+		return params.EnqueuedActions{}, apiservererrors.ServerError(jujuerrors.New("no actions specified"))
 	}
 
-	return "", params.ActionResults{}, errors.NotSupportedf("actions in Dqlite")
-}
+	const leader = "/leader"
+	actionResults := make([]params.ActionResult, len(arg.Actions))
+	actionResultByUnitName := make(map[string]*params.ActionResult)
+	runParams := make([]operation.RunParams, 0, len(arg.Actions))
+	for i, action := range arg.Actions {
+		taskParams := operation.TaskParams{
+			ActionName:     action.Name,
+			Params:         action.Parameters,
+			IsParallel:     action.Parallel == nil || *action.Parallel, // default is parallel
```

Comment:
question: I find parallel defaulting to `true` very odd. Where did you find this? A quick search isn't showing.